public/script.js

```diff
@@ -1,28 +1,32 @@
-$(document).ready(function() {
-    $.ajax({
-        url: '/posts',
-        dataType: 'json',
-        success: function(data) {
+document.onreadystatechange = function () {
+  if (document.readyState === 'complete') {
+    var xhr = new XMLHttpRequest();
+    xhr.onreadystatechange = function() {
+      if(xhr.readyState === 4){
+        if (xhr.status === 200){
+          var data = JSON.parse(xhr.responseText);
+          for (var blogPost in data) {
+            var postDiv         = document.createElement('div');
+            var postText        = document.createElement('p');
+            var thumbnail       = document.createElement('img');
+            var postContainer   = document.getElementsByClassName('post-container')[0];
 
-            for (var blogPost in data) {
-                var postDiv         = document.createElement('div');
-                var postText        = document.createElement('p');
-                var thumbnail       = document.createElement('img');
-                var postContainer   = document.getElementsByClassName('post-container')[0];
+            thumbnail.src = "./img/logo2.png";
+            thumbnail.className = "thumbnail";
+            postText.innerHTML = data[blogPost];
+            postDiv.className = "post";
 
-                thumbnail.src = "./img/logo2.png";
-                thumbnail.className = "thumbnail";
-                postText.innerHTML = data[blogPost];
-                postDiv.className = "post";
-
-                postDiv.appendChild(thumbnail);
-                postDiv.appendChild(postText);
-                postContainer.appendChild(postDiv);
-                
-            }
-        },
-        error: function(error){
-            console.log(error);
+            postDiv.appendChild(thumbnail);
+            postDiv.appendChild(postText);
+            postContainer.appendChild(postDiv);
+          }
+        }
+        else {
+          console.error(JSON.parse(xhr.responseText));
```

Comment:
Another bug :) `responseText` here is a normal string - not stringified JSON - so it breaks when you try to parse it.

The reason why we need to parse [on line 7](https://github.com/node-girls/node-workshop/pull/54/files#diff-a402f43275a83fa1cdaa204ce50036f5R7) is because in that case we're sending back the JSON of our blogposts.  Here though, it's just a regular string!

Reply:
I recommend you test this out in your own code when making this change, so you can see what I mean.  For example, you could [change the endpoint](https://github.com/node-girls/node-workshop/pull/54/files#diff-a402f43275a83fa1cdaa204ce50036f5R29) to something that doesn't exist, then you can see what happens when there's an error.